### PR TITLE
Fix ImageGPT doc example

### LINKED ISSUE #24189: Problems while Running ImageGPT
### System Info

- `transformers` version: 4.28.1
- Platform: Linux-4.15.0-197-generic-x86_64-with-glibc2.27
- Python version: 3.11.3
- Huggingface_hub version: 0.15.1
- Safetensors version: not installed
- PyTorch version (GPU?): 2.0.1 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

_No response_

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

I tried to run the code from the official example: 
https://huggingface.co/docs/transformers/model_doc/imagegpt#transformers.ImageGPTForCausalImageModeling
```python
from transformers import AutoImageProcessor, ImageGPTForCausalImageModeling
import torch
import matplotlib.pyplot as plt
import numpy as np

image_processor = AutoImageProcessor.from_pretrained("openai/imagegpt-small")
model = ImageGPTForCausalImageModeling.from_pretrained("openai/imagegpt-small")
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
model.to(device)

# unconditional generation of 8 images
batch_size = 8
context = torch.full((batch_size, 1), model.config.vocab_size - 1)  # initialize with SOS token
context = torch.tensor(context).to(device)
output = model.generate(
    input_ids=context, max_length=model.config.n_positions + 1, temperature=1.0, do_sample=True, top_k=40
)

clusters = image_processor.clusters
height = image_processor.size["height"]
width = image_processor.size["width"]

samples = output[:, 1:].cpu().detach().numpy()
#Error line below   
samples_img = [
    np.reshape(np.rint(127.5 * (clusters[s] + 1.0)), [height, width, 3]).astype(np.uint8) for s in samples
]  # convert color cluster tokens back to pixels
f, axes = plt.subplots(1, batch_size, dpi=300)

for img, ax in zip(samples_img, axes):
    ax.axis("off")
    ax.imshow(img)
```

Error on line: ```samples_img = [...]``` on the part ```clusters[s]```. 

This is because: 
```samples.shape  = (8,1024)```
At that point, ```s.shape = (1024,)```. 
So, ```Clusters[s]``` cannot index properly. 

**Error Message**:  TypeError: only integer scalar arrays can be converted to a scalar index

### Expected behavior

It should plot 8 different predictions from the ImageGPT Model. 
image

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

There was a bug in the example, where the clusters in the image processor were stored as lists, but the example assumed they were numpy arrays. 

At the moment, clusters are stored as a list of lists, but [converted to a numpy array during processing](https://github.com/huggingface/transformers/blob/0b7b4429c78de68acaf72224eb6dae43616d820c/src/transformers/models/imagegpt/image_processing_imagegpt.py#L230). This PR converts to a numpy array when setting the class attribute and if new `cluster` are passed into `preprocess`. 

This: 
* Maintains backwards compatibility with old configurations
* Saved configs aren't changed (`clusters` is still converted to a list of lists when serializing) see this [dummy image processor](https://huggingface.co/amyeroberts/dummy_imagegpt_image_processor_np_clusters) created from this branch. 
* Is more efficient - we're not converting the same list of lists every batch. 

A potential breaking change is if users were accessing the `clusters` attribute and using it as a list of lists. As this was caught because users were using the clusters as a numpy array (according to the example) I expect this impact to be low. 

Fixes #24189

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/imagegpt/image_processing_imagegpt.py [hunk 1]
```diff
@@ -199,6 +199,7 @@ def preprocess(
         do_normalize = do_normalize if do_normalize is not None else self.do_normalize
         do_color_quantize = do_color_quantize if do_color_quantize is not None else self.do_color_quantize
         clusters = clusters if clusters is not None else self.clusters
+        np.array(clusters)
```
Comment (ydshieh) [inline src/transformers/models/imagegpt/image_processing_imagegpt.py:202]: should assign it to `clusters`, otherwise no effect I believe.

**Review by ydshieh — APPROVED** [review]
Thank you for the doc!

File: src/transformers/models/imagegpt/image_processing_imagegpt.py (hunk 1 above)
Comment (amyeroberts) [reply src/transformers/models/imagegpt/image_processing_imagegpt.py:202]: d'oh 🤦‍♀️

Comment (amyeroberts) [inline src/transformers/models/imagegpt/image_processing_imagegpt.py:202]:
```suggestion
        clusters = np.array(clusters)
```

**Pull request merged**